packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java

```diff
@@ -187,6 +188,19 @@ public Publisher<byte[]> stream(
 		});
 	}
 
+	@GET
+	@Path("/{jobId}/sciml-intermediate-results")
+	@Produces(MediaType.SERVER_SENT_EVENTS)
+	@SseElementType(MediaType.APPLICATION_JSON)
+	@Tag(name = "sciml simulation intermediate results associated with run ID")
+	public Publisher<JsonObject> scimlResult(
+		@PathParam("jobId") final String jobId
+	) {
+		return Multi.createFrom().publisher(scimlQueueStream).filter(event -> {
```

Comment:
(Related to my above comment)
We should specify the object type here to allow us to getId() rather than getValue("id")
This will help us work with this type as well rather than just being a generic object. 
This will also allow us to catch errors if they start sending a different object to us

Reply:
Same reason as above.